var/spack/repos/builtin/packages/py-virtualenv/package.py

```diff
@@ -22,20 +23,28 @@ class PyVirtualenv(PythonPackage):
     version("1.11.6", sha256="3e7a4c151e2ee97f51db0215bfd2a073b04a91e9786df6cb67c916f16abe04f7")
 
     depends_on("python@2.7:2.8,3.4:", type=("build", "run"))
-    depends_on("python@2.7:2.8,3.5:", type=("build", "run"), when="@20.10.0:")
+    depends_on("python@2.7:2.8,3.5:", when="@20.10.0:", type=("build", "run"))
+    depends_on("python@3.6:", when="@20.16.4:", type=("build", "run"))
 
     # not just build-time, requires pkg_resources
     depends_on("py-setuptools@40.6.3:", type=("build", "run"))
-    depends_on("py-setuptools@41.00.03:", type=("build", "run"), when="@20.10.0:")
-    depends_on("py-setuptools-scm@2:", type=("build", "run"), when="@20.10.0:")
+    depends_on("py-setuptools@41.00.03:", when="@20.10.0:", type=("build", "run"))
+    depends_on("py-setuptools-scm@2:", when="@20.10.0:", type=("build", "run"))
 
     depends_on(
-        "py-backports-entry-points-selectable @1.0.4:", type=("build", "run"), when="@20.10.0:"
+        "py-backports-entry-points-selectable @1.0.4:", type=("build", "run"), when="@20.10.0"
     )
-    depends_on("py-distlib@0.3.1:0", type=("build", "run"), when="@20.10.0:")
-    depends_on("py-filelock@3.2:3", type=("build", "run"), when="@20.10.0:")
-    depends_on("py-platformdirs@2:2", type=("build", "run"), when="@20.10.0:")
-    depends_on("py-six@1.9.0:1", type=("build", "run"), when="@20.10.0:")
-    depends_on("py-importlib-metadata@0.12:", type=("build", "run"), when="@20.10.0: ^python@:3.7")
-    depends_on("py-importlib-resources@1:", type=("build", "run"), when="@20.10.0: ^python@:3.6")
-    depends_on("py-pathlib2", type=("build", "run"), when="@20.10.0: ^python@:3.3")
+    depends_on("py-distlib@0.3.1:0", when="@20.10.0:", type=("build", "run"))
+    depends_on("py-distlib@0.3.5:0", when="@20.16.4:", type=("build", "run"))
+    depends_on("py-filelock@3.2:3", when="@20.10.0:", type=("build", "run"))
+    depends_on("py-filelock@3.4.1:3", when="@20.16.4:", type=("build", "run"))
+    depends_on("py-platformdirs@2:2", when="@20.10.0:", type=("build", "run"))
+    depends_on("py-platformdirs@2.4:2", when="@20.16.4:", type=("build", "run"))
+    depends_on("py-six@1.9.0:1", when="@20.10.0", type=("build", "run"))
+    depends_on("py-importlib-metadata@0.12:", when="@20.10.0: ^python@:3.7", type=("build", "run"))
+    depends_on("py-importlib-resources@1:", when="@20.10.0: ^python@:3.6", type=("build", "run"))
+    depends_on(
+        "py-importlib-metadata@4.8.3:", when="@20.10.0: ^python@:3.7", type=("build", "run")
+    )
+    depends_on("py-importlib-resources@5.4:", when="@20.10.0: ^python@:3.6", type=("build", "run"))
```

Comment:
I wouldn't expect these `when` versions to be identical, did you mean `@20.16.4:`?